From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Wed, 17 Aug 2016 20:36:51 +0200
Subject: [PATCH] build: cache: Consider UBUNTU_ARCH in .cacheid

Usually UBUNTU_ARCH is not changed inside the same repository,
but we should consider this case anyway for future merging
Armband and Fuel@OPNFV repositories/build systems.

This applies to all Fuel plugins and the local repo.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 build/f_isoroot/f_bgpvpn-pluginbuild/Makefile              | 1 +
 build/f_isoroot/f_collectd-ceilometer-pluginbuild/Makefile | 1 +
 build/f_isoroot/f_kvm-pluginbuild/Makefile                 | 1 +
 build/f_isoroot/f_odlpluginbuild/Makefile                  | 1 +
 build/f_isoroot/f_onosfwpluginbuild/Makefile               | 1 +
 build/f_isoroot/f_ovs-nsh-dpdk-pluginbuild/Makefile        | 1 +
 build/f_isoroot/f_ovsnfv-dpdk-pluginbuild/Makefile         | 1 +
 build/f_isoroot/f_repobuild/Makefile                       | 1 +
 build/f_isoroot/f_vsperfpluginbuild/Makefile               | 1 +
 build/f_isoroot/f_yardstick-pluginbuild/Makefile           | 1 +
 10 files changed, 10 insertions(+)

diff --git a/build/f_isoroot/f_bgpvpn-pluginbuild/Makefile b/build/f_isoroot/f_bgpvpn-pluginbuild/Makefile
index ed3be29..8ce28fc 100644
--- a/build/f_isoroot/f_bgpvpn-pluginbuild/Makefile
+++ b/build/f_isoroot/f_bgpvpn-pluginbuild/Makefile
@@ -67,6 +67,7 @@ release:.bgpvpnbuild
 	fi
 	sha1sum Makefile >> .cachedata
 	sha1sum config.mk >> .cachedata
+	echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_collectd-ceilometer-pluginbuild/Makefile b/build/f_isoroot/f_collectd-ceilometer-pluginbuild/Makefile
index 9ab922d..17bc36d 100644
--- a/build/f_isoroot/f_collectd-ceilometer-pluginbuild/Makefile
+++ b/build/f_isoroot/f_collectd-ceilometer-pluginbuild/Makefile
@@ -67,6 +67,7 @@ release:.fastpathmetrics
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	@cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_kvm-pluginbuild/Makefile b/build/f_isoroot/f_kvm-pluginbuild/Makefile
index ba2fd0b..2454b5e 100644
--- a/build/f_isoroot/f_kvm-pluginbuild/Makefile
+++ b/build/f_isoroot/f_kvm-pluginbuild/Makefile
@@ -70,6 +70,7 @@ release:.kvmbuild
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	@cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_odlpluginbuild/Makefile b/build/f_isoroot/f_odlpluginbuild/Makefile
index e2cbf7d..6c717c7 100644
--- a/build/f_isoroot/f_odlpluginbuild/Makefile
+++ b/build/f_isoroot/f_odlpluginbuild/Makefile
@@ -58,6 +58,7 @@ release:.odlbuild
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_onosfwpluginbuild/Makefile b/build/f_isoroot/f_onosfwpluginbuild/Makefile
index a9dcb22..02ba029 100644
--- a/build/f_isoroot/f_onosfwpluginbuild/Makefile
+++ b/build/f_isoroot/f_onosfwpluginbuild/Makefile
@@ -66,6 +66,7 @@ release:.onosbuild
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	@cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_ovs-nsh-dpdk-pluginbuild/Makefile b/build/f_isoroot/f_ovs-nsh-dpdk-pluginbuild/Makefile
index ecf586a..2d0d202 100644
--- a/build/f_isoroot/f_ovs-nsh-dpdk-pluginbuild/Makefile
+++ b/build/f_isoroot/f_ovs-nsh-dpdk-pluginbuild/Makefile
@@ -67,6 +67,7 @@ release:.ovsbuild
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	@cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_ovsnfv-dpdk-pluginbuild/Makefile b/build/f_isoroot/f_ovsnfv-dpdk-pluginbuild/Makefile
index 9e432d1..9455389 100644
--- a/build/f_isoroot/f_ovsnfv-dpdk-pluginbuild/Makefile
+++ b/build/f_isoroot/f_ovsnfv-dpdk-pluginbuild/Makefile
@@ -67,6 +67,7 @@ release:.ovsnfvbuild
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	@cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_repobuild/Makefile b/build/f_isoroot/f_repobuild/Makefile
index 6fef8ff..f6ebc2c 100644
--- a/build/f_isoroot/f_repobuild/Makefile
+++ b/build/f_isoroot/f_repobuild/Makefile
@@ -73,6 +73,7 @@ release:nailgun
 	sha1sum Makefile >> .cachedata
 	sha1sum config.mk >> .cachedata
 	$(CACHETOOL) packages >> .cachedata
+	echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_vsperfpluginbuild/Makefile b/build/f_isoroot/f_vsperfpluginbuild/Makefile
index 9bd236a..73de611 100644
--- a/build/f_isoroot/f_vsperfpluginbuild/Makefile
+++ b/build/f_isoroot/f_vsperfpluginbuild/Makefile
@@ -68,6 +68,7 @@ release:.vsperfbuild
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	@cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
diff --git a/build/f_isoroot/f_yardstick-pluginbuild/Makefile b/build/f_isoroot/f_yardstick-pluginbuild/Makefile
index 47723d5..32de3b3 100644
--- a/build/f_isoroot/f_yardstick-pluginbuild/Makefile
+++ b/build/f_isoroot/f_yardstick-pluginbuild/Makefile
@@ -68,6 +68,7 @@ release:.yardstickbuild
 	fi
 	@sha1sum Makefile | awk {'print $$1'} >> .cachedata
 	@sha1sum config.mk | awk {'print $$1'} >> .cachedata
+	@echo -n $(UBUNTU_ARCH) | sha1sum | awk {'print $$1'} >> .cachedata
 	@cat .cachedata | $(CACHETOOL) getid > .cacheid
 
 # Clean local data related to caching - called prior to ordinary build
